### PR TITLE
[4][com_actionlog] - log event reset password

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #39062.

### Summary of Changes
generate proper event

### Testing Instructions

- Set up sending mail from the site,
- Create a test user with a valid email address,
- Use the item "Forgot your passwor" in the authorization form in the frontend for a test user,
- Enter your login / confirmation code or follow the link in the letter received by mail,
- Change the user's password to a new one.

### Actual result BEFORE applying this Pull Request

see #39435 

### Expected result AFTER applying this Pull Request
the events are correctly logged in Action Logged

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by N6REJ** [comment]
please fill out PR info properly.

File: plugins/actionlog/joomla/src/Extension/Joomla.php [hunk 1]
```diff
@@ -1127,4 +1123,70 @@ private function getActionLogParams($context): ?stdClass
...
+            'accountlink' => 'index.php?option=com_users&task=user.edit&id=' . $user->id,
+        );
+
+        $this->addLog(array($message), 'PLG_ACTIONLOG_JOOMLA_USER_RESET_REQUEST', $context, $user->id);
+    }
+
+    /**
+     * On after Completed reset request
+     *
+     * Method is called after user request to reset their password.
```
Comment (brianteeman) [inline plugins/actionlog/joomla/src/Extension/Joomla.php]: copy pasted from previous function

Comment (alikon) [reply plugins/actionlog/joomla/src/Extension/Joomla.php]: fixed

**Comment by sandewt** [comment]
> Pull Request for Issue https://github.com/joomla/joomla-cms/pull/39435 .
see https://github.com/joomla/joomla-cms/pull/39435

Shoud be #39062

**Comment by SharkyKZ** [comment]
Surely, adding new events is not necessary to fix the reported issue?

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on fa806962ad5dc8b0310c794e129e35d5f1225d2e<br /><br />Thanks, now it works as it should.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39435">issues.joomla.org/tracker/joomla-cms/39435</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on fa806962ad5dc8b0310c794e129e35d5f1225d2e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39435">issues.joomla.org/tracker/joomla-cms/39435</a>.</sub>

**Comment by richard67** [comment]
Hmm there is a comment with a valid concern: https://github.com/joomla/joomla-cms/pull/39435#issuecomment-1356111829 . We should not add new events when it‘s not needed.

**Comment by alikon** [comment]
What's the scope of the Events then ?

**Comment by richard67** [comment]
> What's the scope of the Events then ?

@alikon Have you checked if there is an existing event which can be used? If you say yes and there was none, I am ok with it.

**Comment by alikon** [comment]
in the previous version 3 it was used  `onUserAfterSave` but 1st is not logical 2nd no separation of concerns, and 3rd not readable code

File: components/com_users/src/Model/ResetModel.php [hunk 2]
```diff
@@ -468,9 +477,17 @@ public function processResetRequest($data)
         // Check for an error.
         if ($return !== true) {
             return new \Exception(Text::_('COM_USERS_MAIL_FAILED'), 500);
-        } else {
-            return true;
         }
+
+        $event = AbstractEvent::create(
+            'onUserAfterReset',
```
Comment (rdeutz) [inline components/com_users/src/Model/ResetModel.php]: I think onUserAfterResetRequest would be a better name

**Comment by rdeutz** [comment]
I think it is fine to create new events for thinks like this, but it should be consisted overt the whole process

File: components/com_users/src/Model/ResetModel.php (hunk 2 above)
Comment (alikon) [reply components/com_users/src/Model/ResetModel.php]: changed

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on ba1d35053f4f35afa3df196eb91bc29d9c02482d<br /><br />I re-tested after the changes - everything works. Thank you.
I hope this doesn't get stuck and gets merged.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39435">issues.joomla.org/tracker/joomla-cms/39435</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on ba1d35053f4f35afa3df196eb91bc29d9c02482d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39435">issues.joomla.org/tracker/joomla-cms/39435</a>.</sub>

**Comment by richard67** [comment]
> I think it is fine to create new events for thinks like this, but it should be consisted overt the whole process

@alikon If I understand @rdeutz 's comment right, he means that if we add events `onUserAfterResetRequest` and `onUserAfterResetComplete`, there should also be events `onUserBeforeResetRequest` and `onUserBeforeResetComplete` added, even if currently not used. @rdeutz Is my understanding right, and should this be done with this PR here?

**Comment by richard67** [comment]
@alikon Could you check the previous two comments and if possible implement the requested changes? It would be good to bring this useful PR forward. Let us know if you need some help or other that or further advice. Thanks in advance.

**Comment by alikon** [comment]
added the `Before` events

**Comment by Kostelano** [comment]
I tested again. At the moment, the PR is broken, something needs to be fixed (problem after the last changes). A request to remind the login comes to the mail, a request to change the password does not. I tried several times.

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on afc06a4f0561d9fac5cfac0583e4a6f28d4a165c<br /><br />Sorry, it works. Sending a successful test.

Why did it happen that at first it did not work: I created a user and, in addition to the "Registered" group, added him to the Superusers group. The system does NOT send a password change request for a group of Superusers (which is strange because I have seen error lines for superusers who want to change the password). In this case, the system proceeds to the next stage - entering a login / confirmation code, but nothing comes to the mail. Well, that's a bug for a separate fix.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39435">issues.joomla.org/tracker/joomla-cms/39435</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on afc06a4f0561d9fac5cfac0583e4a6f28d4a165c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39435">issues.joomla.org/tracker/joomla-cms/39435</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on afc06a4f0561d9fac5cfac0583e4a6f28d4a165c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39435">issues.joomla.org/tracker/joomla-cms/39435</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39435">issues.joomla.org/tracker/joomla-cms/39435</a>.</sub>

**Comment by sandewt** [comment]
> Why don't you use [ ] (brackets) instead of array() ?
See also line 1179.

@alikon Did I miss something, because as far as I know brackets are preferred in J4? You probably have a reason for it.

**Comment by roland-d** [comment]
Thank you

**Comment by alikon** [comment]
@sandewt i'll change it to `[]` on a next pr

**Pull request merged**